### PR TITLE
DCOS-10991 [1 of 4] Add LocalVolumes reducer

### PR DESCRIPTION
**Pull request by Poltergeist**
This adds the local Volumes reducer which is part
of DCOS-10991

### CODE REVIEW

File: plugins/services/src/js/reducers/serviceForm/LocalVolumes.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+         * 2) Set the size from `volume.persistent.size`on the path
+         *    `localVolumes.${index}.size`.
+         * 3) Set the containerPath from `volume.containerPath on the path
+         *    `localVolumes.${index}.containerPath`
+         * 4) Set the mode from `volume.mode` on the path
+         *    `localVolumes.${index}.mode`
+         */
+        memo.push(new Transaction(['localVolumes'], index, ADD_ITEM));
+
+        if (item.persistent != null && item.persistent.size != null) {
```
Comment (nLight) [inline plugins/services/src/js/reducers/serviceForm/LocalVolumes.js]: `item.persistent != null` is it redundant? I think we're already filtering `nulls`

Comment (Poltergeist) [inline plugins/services/src/js/reducers/serviceForm/LocalVolumes.js]: You are right.

File: plugins/services/src/js/reducers/serviceForm/LocalVolumes.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  },
+
+  FormReducer(state = [], {type, path, value}) {
+    if (path == null) {
+      return state;
+    }
+
+    let joinedPath = path.join('.');
+
+    if (joinedPath.search('localVolumes') !== -1) {
```
Comment (wavesoft) [inline plugins/services/src/js/reducers/serviceForm/LocalVolumes.js]: What about `path[0] === 'localVolumes'` ?

File: plugins/services/src/js/reducers/serviceForm/LocalVolumes.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            state = state.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return state;
+      }
+
+      let index = joinedPath.match(/\d+/)[0];
```
Comment (wavesoft) [inline plugins/services/src/js/reducers/serviceForm/LocalVolumes.js]: What about `index = path[1]`?

**Comment by nLight** [comment]
retest this please 😩

**Comment by nLight** [comment]
retest this please :godmode:

**Comment by nLight** [comment]
retest this please 🎉

**Comment by nLight** [comment]
\o/

**Pull request merged**